packages/core/src/components/templates/FieldTemplate/FieldTemplate.tsx

```diff
@@ -22,17 +21,25 @@ export default function FieldTemplate<T = any, F = any>(
     hidden,
     required,
     displayLabel,
+    registry,
+    uiSchema,
   } = props;
+  const uiOptions = getUiOptions(uiSchema);
+  const WrapIfAdditionalTemplate = getTemplate(
```

Comment:
Adding the string name as the generic will ensure the proper typing
```suggestion
  const WrapIfAdditionalTemplate = getTemplate<"WrapIfAdditionalTemplate">(
```